### PR TITLE
Fixes #28226 - upgrade to sprockets 4

### PR DESCRIPTION
**Pull request by ezr-ondrej**
- [x] Prepare plugin foreman-tasks - theforeman/foreman-tasks#478
- [x] rebase after merge of #7191
- [ ] try to build

### CODE REVIEW

File: bundler.d/assets.rb
Comment (ezr-ondrej) [inline bundler.d/assets.rb:7]: This becomes wrapper for `sassc-rails` - for some reason we can't use `sassc-rails` directly :man_shrugging:

File: config/initializers/assets.rb
Comment (ezr-ondrej) [inline config/initializers/assets.rb:26]: Do someone has an idea what sorcery is hidden in this block?

File: app/assets/config/manifest.js
Comment (ezr-ondrej) [inline app/assets/config/manifest.js:12]: Probably forgotten file in precompiles, as it doesn't exists, but I've added it to be noticed it's not compiled now, do someone know if it is safe to remove?
(same goes for `reports.js`, `gettext/all.js`)

File: config/initializers/assets.rb
Comment (ekohl) [reply config/initializers/assets.rb:26]: Probably because every plugin brings its own manifest.json file. Most Rails deployments are monolithic where one build is made and compiled. In our setup we explicitly build a core  and then build plugins. Each plugins ships its own assets and a manifest. By default sprockets doesn't work this way so we have this code.

**Review by ekohl — COMMENTED** [review]
A historical note: every time we made changes to our asset handling, we broke things with plugins. That makes me cautious.

It mentions the new version ships integration with babel. If it bundles a copy of babel I'm a bit concerned about that. It makes me wonder how hard it would be to move more things to webpack. While I don't like webpack, I do like a single way of handling things.

**Comment by mmoll** [comment]
Needs to wait until Ruby 2.3 support is dropped.

File: bundler.d/assets.rb
Comment (mmoll) [reply bundler.d/assets.rb:7]: Related: https://github.com/theforeman/rails-packaging/pull/153 - that update could get into develop unrelated to the sprockets update, but we need to wait until the RPM repos are merged.

File: bundler.d/assets.rb
Comment (mmoll) [inline bundler.d/assets.rb]: Can this (or even 6.x) get in without the sprockets update?

File: bundler.d/assets.rb
Comment (mmoll) [inline bundler.d/assets.rb]: Can this get in without the sprockets update?

Comment (ezr-ondrej) [reply bundler.d/assets.rb]: yeah sure, I'll split it.

**Comment by ezr-ondrej** [comment]
I've extracted the assets upgrade: #7191.

**Review by ekohl — DISMISSED** [review]
From a packaging perspective, :+1: since https://github.com/theforeman/foreman-packaging/commit/b0018b43e451fc1a9c5afde2b7ffd612874db894 was already merged. So we either need to revert that or merge this.

Haven't looked at the actual code.

**Comment by ezr-ondrej** [comment]
I was testing this with all the plugins, trying to read code for consequences, so I believe it should be safe, it's quite some time ago and stabilization is next week. As we don't need this, I'd revert packaging. Thoughts?

**Comment by mmoll** [comment]
I tested this a while ago when it was fresh, so I'm all in...

**Comment by mmoll** [comment]
@ezr-ondrej could you please rebase? Katello should turn green then...

**Comment by ezr-ondrej** [comment]
`Everything up-to-date` :(

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by mmoll** [comment]
I guess Katello/Bastion/Tests need adjustment :disappointed:

**Comment by mmoll** [comment]
Needs https://github.com/Katello/katello/pull/8764 and RPM packaging, I guess.

**Comment by ezr-ondrej** [comment]
[test katello]

**Comment by ezr-ondrej** [comment]
The https://github.com/Katello/katello/pull/8756 could have done the job already :pray:

**Comment by jlsherrill** [comment]
[test katello]

**Comment by mmoll** [comment]
RPM PR: https://github.com/theforeman/foreman-packaging/pull/5404

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, díky @ezr-ondrej!

File: app/assets/config/manifest.js
Comment (tbrisker) [reply app/assets/config/manifest.js:12]: I believe these are all leftovers that were moved to webpack in the past and can be dropped from here.

**Pull request merged**
